crates/sui-sdk/examples/utils/example_utils.rs

```diff
@@ -0,0 +1,180 @@
+
+    let coin_obj_id = coin_with_sui(&client, active_address).await?;
+    Ok((client, active_address, *recipient, coin_obj_id))
+}
+
+/// Return a sui client to interact with the APIs, and the active address.
+pub(crate) async fn setup_for_read() -> Result<(SuiClient, SuiAddress), anyhow::Error> {
+    let client = SuiClientBuilder::default().build_testnet().await?;
+    println!("Sui testnet version is: {}", client.api_version());
+    let mut wallet = retrieve_wallet().await?;
+    assert!(wallet.get_addresses().len() >= 2);
+    let active_address = wallet.active_address()?;
+
+    Ok((client, active_address))
+}
+
+/// Return the coin object ID with at least 100_000_000 MIST owned by the provided active address.
+pub(crate) async fn coin_with_sui(
+    client: &SuiClient,
+    active_address: SuiAddress,
+) -> Result<ObjectID, anyhow::Error> {
+    let coin_type = "0x2::sui::SUI".to_string();
+
+    let coins = client
+        .coin_read_api()
+        .get_coins_stream(active_address, Some(coin_type.clone()))
+        .filter(|coin| future::ready(coin.balance >= 100_000_000))
+        .collect::<Vec<_>>()
+        .await;
+
+    if coins.is_empty() {
+        request_tokens_from_faucet(active_address).await?;
+    }
+
+    let coins = client
+        .coin_read_api()
+        .get_coins_stream(active_address, Some(coin_type.clone()))
+        .filter(|coin| future::ready(coin.balance >= 100_000_000))
+        .collect::<Vec<_>>()
+        .await;
+    let coin_obj_id = coins.get(0).unwrap().coin_object_id;
+    return Ok(coin_obj_id);
+}
+
+/// Request tokens from the Faucet for the given address
+pub(crate) async fn request_tokens_from_faucet(address: SuiAddress) -> Result<(), anyhow::Error> {
+    let address_str = address.to_string();
+    let json_body = json![{
+        "FixedAmountRequest": {
+            "recipient": &address_str
+        }
+    }];
+
+    // make the request to the faucet JSON RPC API for coin
+    let client = Client::new();
+    let resp = client
+        .post(SUI_FAUCET)
+        .header("Content-Type", "application/json")
+        .json(&json_body)
+        .send()
+        .await?;
+    println!("Faucet request status: {}", resp.status());
+    // wait for a bit before jumping to other methods
+    std::thread::sleep(std::time::Duration::from_secs(2));
+    Ok(())
+}
+
+/// Return a transaction digest from a split coin + merge coins transaction
+pub(crate) async fn split_coin_digest(
+    sui: &SuiClient,
+    sender: &SuiAddress,
+) -> Result<TransactionDigest, anyhow::Error> {
+    let coin_type = "0x2::sui::SUI".to_string();
+    let coins = sui
+        .coin_read_api()
+        .get_coins(*sender, Some(coin_type), None, None)
+        .await?;
+
+    let coins = coins.data;
+    let coins_len = coins.len();
+    let coin_for_split = coins
+        .into_iter()
+        .filter(|x| x.balance >= 500_000)
+        .collect::<Vec<_>>();
+
+    let coin = coin_for_split
+        .get(0)
```

Comment:
Instead of collecting into a vector and then getting the first element, you can just call `next` on the iterator to get the first element.